applications/crossbar/src/modules/cb_ledgers.erl

```diff
@@ -0,0 +1,208 @@
+-spec resource_exists() -> 'true'.
+-spec resource_exists(path_token()) -> 'true'.
+resource_exists() -> 'true'.
+resource_exists(_) -> 'true'.
+resource_exists(_, ?CREDIT) -> 'true';
+resource_exists(_, ?DEBIT) -> 'true'.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec authenticate(cb_context:context()) -> boolean().
+authenticate(_Context) -> 'false'.
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec authorize(cb_context:context()) -> boolean().
+authorize(_Context) -> 'false'.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Check the request (request body, query string params, path tokens, etc)
+%% and load necessary information.
+%% /ledgers mights load a list of ledgers objects
+%% /ledgers/123 might load the ledgers object 123
+%% Generally, use crossbar_doc to manipulate the cb_context{} record
+%% @end
+%%--------------------------------------------------------------------
+-spec validate(cb_context:context()) -> cb_context:context().
+-spec validate(cb_context:context(), path_token()) -> cb_context:context().
+validate(Context) ->
+    validate_ledgers(Context, cb_context:req_verb(Context)).
+
+validate(Context, Id) ->
+    validate_ledger(Context, Id, cb_context:req_verb(Context)).
+
+validate(Context, Id, _Action) ->
+    validate_ledger(Context, Id, cb_context:req_verb(Context)).
+
+
+validate_ledgers(Context, ?HTTP_GET) ->
+    read_ledgers(Context).
+
+validate_ledger(Context, Id, ?HTTP_GET) ->
+    read_ledger(Context, Id);
+validate_ledger(Context, _Id, ?HTTP_POST) ->
+    cb_context:validate_request_data(<<"ledgers">>, Context, fun is_admin/1).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% If the HTTP verb is POST, execute the actual action, usually a db save
+%% (after a merge perhaps).
+%% @end
+%%--------------------------------------------------------------------
+-spec post(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+post(Context, Id, ?CREDIT) ->
+    credit_or_debit(Context, Id, ?CREDIT);
+post(Context, Id, ?DEBIT) ->
+    credit_or_debit(Context, Id, ?DEBIT).
+
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+credit_or_debit(Context, Name, Action) ->
+    Fun = wh_util:to_atom(Action, 'true'),
+    AccountId = cb_context:account_id(Context),
+    ReqData = cb_context:req_data(Context),
+    Amount = wh_json:get_integer_value(<<"amount">>, ReqData, 0),
+    Desc = wh_json:get_ne_binary_value(<<"description">>, ReqData),
+    case kz_ledger:Fun(Name, Amount, AccountId, Desc) of
```

Comment:
Better yet, call `case process_action(Action, Name, Amount, AccountId, Desc) of` here and define `process_action` as:

```erlang
process_action(?CREDIT, Name, Amount, AccountId, Desc) ->
    kz_ledger:credit(Name, Amount, AccountId, Desc);
...
```

Dynamic dispatching comes at a cost, and for only two methods, this doesn't seem like a win.